client/matchmaking/matchmaking-preferences-reducer.ts

```diff
@@ -21,7 +20,7 @@ export class MatchmakingPreferencesRecord extends ImmutableRecord({
   race: 'r' as RaceChar,
   mapPoolId: 1,
   mapPoolOutdated: false,
-  mapSelections: List<MapInfoJson>(),
+  mapSelections: List<typeof MapRecord>(),
```

Comment:
don't think this is the right typing, You probably want `ReturnType<typeof MapRecord>` but a better thing to do here might be to just make `MapRecord` do the `extends Record` thing we do elsewhere so that you can use it as a type directly. I don't think it should change other usages of the type very much (or it will at the very least be easy to find with the TS compiler)

Reply:
Hmm, looks like this is not that easy without full TS-ification of the maps reducer, I think. When `MapRecord` extends `Record`, the typing here works, but then creation of the map record fails typecheck, as we're trying to create a `MapRecord` from `MapInfoJson`. Not really sure how to fix that, but I also saw that `maps2-reducer` doesn't even use the `MapRecord`, so maybe when we fully move to that it won't even matter?

Reply:
yes, maps2-reducer is immer so these things are simpler. I don't see why creating a MapRecord from a MapInfoJson would be a problem here, could you link me to the code you're talking about? If there is a typing mismatch on the fields then MapRecord is probably just wrong, MapInfoJson is as accurate to what we actually store/pass as anything is (and with MapRecord I spent a bit less time trying to get things right).

If it *needs* to differ, you could also add a constructor/method on MapRecord to convert from a MapInfoJson until we can deleted it entirely.